Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -203,15 +203,20 @@ import Foundation
     /// Fetches a customer's vaulted payment method nonces.
     /// Must be using client token with a customer ID specified.
     ///  - Parameter completion: Callback that returns either an array of payment method nonces or an error
+    ///  - Note: Only the top level `BTPaymentMethodNonce` type is returned, fetching any additional details will need to be done on the server
     public func fetchPaymentMethodNonces(_ completion: @escaping ([BTPaymentMethodNonce]?, Error?) -> Void) {
         fetchPaymentMethodNonces(false, completion: completion)
     }
 
+    // NEXT_MAJOR_VERSION: this should move into the Drop-in for parity with Android
+    // This will also allow us to return the types directly which we were doing in the +load method
+    // previously in Obj-C - this is not available in Swift
```

Comment:
We've actually gotten multiple requests to move this back into core. We initially moved it to DropIn to get rid of circular dependencies, and we're planning to implement it the correct way in `v5`.

Reply:
Ah interesting... we may have to rethink what we want to return to merchants if we move it back to core. Not sure if Android is capable of what we were doing in v5. We were using the `load` method to dynamically register types that don't exist in core (`BTCardNonce`/`PayPalNonce`/'VenmoNonce`/`ApplePayNonce`) at compile time. We don't have the ability to do that hacky workaround in Swift and importing those modules into Core would cause a circular dependency. How are y'all planning to correctly implement it in v5?

Reply:
Yeah it was a hack in Android `v3` as well. We used `compileOnly` in our Gradle files to basically create weak references in the dependency tree. That allowed Gradle to build successfully, but it caused some issues in the long run.

By "correct" I'm thinking "in a non-hacky way." It's tough because there will probably be tradeoffs. An enum feels like the right way to me, but I'd be up to do some cross-platform pairing on this since we need a solution on Android too.